apps/HowtoAddNewApp.md

```diff
@@ -0,0 +1,185 @@
+# How to add new application
+
+Five files, **source file**, [**Kconfig**](#kconfig), [**Kconfig_ENTRY**](#kconfig_entry), [**Make.defs**](#makedefs) and [**Makefile**](#makefile) are mandatory to add a new application.
+
+## Kconfig
+This file contains configurations of new application to enable or to set specific value.  
+Mandatory is adding two configs like shown below:
+```
+config <APP_NAME>
+	bool "<Write Description>"
+	default n
+	---help---
+		<Write Details>
+
+config USER_ENTRYPOINT
+	string
+	default "<APP_MAIN_FUNCTION>" if ENTRY_<APP_NAME>
+```
+First controls enabling application and second provides a main function pointer for CONFIG_USER_ENTRYPOINT.  
```

Comment:
The first config parameter specifies whether the application should be built or not.
The second config parameter specifies the function name of the application which should be executed when OS executes the application. The entrypoint can be set manually but for it the name of function which has to be executed firstly should be known for each application. This option registers the app with the build system and can be selected from the menu to be used as application entrypoint.